test/container_app_test.py

```diff
@@ -65,6 +67,41 @@ async def test_container_snapshot_restore(container_client, tmpdir, servicer):
         assert old_client.credentials == ("ta-i-am-restored", "ts-i-am-restored")
 
 
+@pytest.mark.asyncio
+async def test_container_snapshot_restore_heartbeats(tmpdir, servicer):
+    client = _Client(servicer.container_addr, api_pb2.CLIENT_TYPE_CONTAINER, ("ta-123", "task-secret"))
+    async with client as async_client:
+        io_manager = _ContainerIOManager(api_pb2.ContainerArguments(), async_client)
+        restore_path = tmpdir.join("fake-restore-state.json")
+        restore_path.write_text(
+            json.dumps(
+                {
```

Comment:
nit: think we can break out this repeated string into a hlper constant now to save some lines.